### PR TITLE
EVNT-35 Refactor the code in the interceptors to a utility class

### PR DESCRIPTION
**Pull request by hassam303**
I refactored the require interceptor classes and made some additions to pom.xml files to make sure the new superclass will build.

### CODE REVIEW

**Comment by hassam303** [comment]
I had issues building the test suite for the fix. I also experienced these errors when I tried to build the event module without making any modifications.

**Comment by reubenvarghese1** [comment]
@hassam303 Could you please squash your commits as per [https://wiki.openmrs.org/plugins/servlet/mobile#content/view/60785449](url) ?

File: api/src/main/java/org/openmrs/event/HibernateEventInterceptorSuperClass.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+import org.openmrs.event.Event.Action;
+
+/**
+ * A hibernate {@link Interceptor} implementation, intercepts any database inserts, updates and
+ * deletes in a single hibernate session and fires the necessary events. Any changes/inserts/deletes
+ * made to the DB that are not made through the application won't be detected by the module.
+ *
+ * We use a Stack here to handle any nested transactions that may occur within a single thread
+ */
+public abstract class HibernateEventInterceptorSuperClass extends EmptyInterceptor {
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/event/HibernateEventInterceptorSuperClass.java]: Rename this to `BaseHibernateEventInterceptor`

File: api/pom.xml [hunk 2]
```diff
@@ -114,6 +114,14 @@
 				</exclusions>
 			</dependency>	
 
+			<!-- https://mvnrepository.com/artifact/org.hibernate/hibernate-core -->
```
Comment (wluyima) [inline api/pom.xml]: Why are you adding this?

**Comment by wluyima** [comment]
I think we could also merge the two interceptor back into 1 as before and load the appropriate Persistent class object in the onCollectionUpdate() method

File: api/pom.xml (hunk 2 above)
Comment (hassam303) [inline api/pom.xml]: Without this dependency added I was unable to build the files in the Event Module API folder. I've included a screenshot of the error I receive. 
screen shot 2017-03-30 at 9 30 25 am

File: api/src/main/java/org/openmrs/event/HibernateEventInterceptorSuperClass.java (hunk 1 above)
Comment (hassam303) [inline api/src/main/java/org/openmrs/event/HibernateEventInterceptorSuperClass.java]: Done!

File: api/pom.xml (hunk 2 above)
Comment (wluyima) [inline api/pom.xml]: I get that but that's not correct solution for the compilation issue. FYI you don't need this if you redo these interceptors as I suggested in my comments i.e have one interceptor as we did before it was split into 2, load the old Persistent class, if it fails load the new one and then use reflection to call the getOwner() method.

File: api/src/main/java/org/openmrs/event/HibernateEventInterceptorSuperClass.java (hunk 1 above)
Comment (wluyima) [inline api/src/main/java/org/openmrs/event/HibernateEventInterceptorSuperClass.java]: Did you see my comment about merging these interceptors back into one interceptor? And use reflection to call getOwner() after loading the appropriate class

**Comment by hassam303** [comment]
@wluyima Thank you for your active help but I need a bit more clarification. Let me make sure I am understanding. 
1. You would like me to merge the two "HibernateEventInterceptor" subclasses together, declare an abstract getOwner() method in the Base class, and implement it in the NEW singular MERGED "HibernateEvenInterceptor" subclass? 

2. Is the base class not already in the API subproject? Current path: 
(openmrs-module-event/api/src/main/java/org/openmrs/event)

Thanks again for your help!

**Comment by wluyima** [comment]
Historically, there used to be one interceptor named `HibernateEventInterceptor`, after we upgraded to hibernate 4 in core the interceptor was split into 2 to support older and newer versions of OpenMRS core that were running different hibernate versions since PersistentCollection class had been moved to a new package in hibernate 4, this wasn't a proper solution in my opinion, what am suggesting is that we go back to a single `HibernateEventInterceptor` class and re-implement the `onCollectionUpdate()` method to load the proper instance of the PersistentCollection class and use reflection to invoke the getOwner() method

**Comment by dkayiwa** [comment]
@hassam303 are you still working on this?

**Comment by dkayiwa** [comment]
@hassam303 ping me to reopen when you resume working on this.

**Pull request closed without merging**